include/column-list-table.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * column-list-table.h - helper functions for implementing -H/--list-columns option
+ *
+ * Copyright (C) 2023 Red Hat, Inc. All rights reserved.
+ * Written by Masatake YAMATO <yamato@redhat.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it would be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef UTIL_LINUX_COLUMN_LIST_TABLE_H
+#define UTIL_LINUX_COLUMN_LIST_TABLE_H
+
+#include "libsmartcols.h"
+#include <stdio.h>
+
+enum { CLT_COL_HOLDER, CLT_COL_TYPE, CLT_COL_DESC };
+
+static struct libscols_table *xcolumn_list_table_new(const char *table_name,
+						     FILE *out,
+						     int raw,
+						     int json)
+{
+	struct clt_colinfo {
+		const char *name;
+		int flags;
+	};
+	const struct clt_colinfo clt_infos[] = {
+		[CLT_COL_HOLDER] = { "HOLDER", SCOLS_FL_RIGHT },
+		[CLT_COL_TYPE]   = { "TYPE",   0 },
+		[CLT_COL_DESC]   = { "DESC",   0 },
+	};
```

Comment:
Data-based programming is the best, but I think `struct clt_colinfo` is overkill in this case. All you need is to call scols_table_new_column() three times.
```C
if (!scols_table_new_column(tb, "HOLDER", 0, SCOLS_FL_RIGHT))
  goto failed;
if (!scols_table_new_column(tb, "TYPE", 0, 0))
  goto failed;
if (!scols_table_new_column(tb, "DESCRIPTION", 0, 0))
  goto failed;
```
you do not need scols_column_set_json_type(cl, SCOLS_JSON_STRING); the string is default.